### DISCUSSION TITLE
Finding Context for Music-Specific JSON-LD Keys

### DISCUSSION THREAD

**Discussion by malajvan** [DDMAL/linkedmusic-datalake#10] [General]
I want to discuss the problem of providing context for music-specific features in JSON-LD. These features are the ones we deemed too domain-specific for Wikidata. 
We have already encountered this while working with simssadb and cantusdb (Finalis and Last Pitch Class Name feature). You can review the discussion here: [Discussion Link](https://github.com/DDMAL/linkedmusic-datalake/discussions/8#discussioncomment-6918586).

I thought of potential approaches to handle the context of music-specific features in our JSON-LD representations:

1. **Do Not Link Them:** Under this proposal, we would not provide a context for music-specific terms in our JSON-LD data. The result will not align with Linked Data principles (see [here](https://www.w3.org/TR/json-ld/#iris)). JSON-LD processors and algorithms will ignore these keys when processing the data, and the information will not be considered Linked Data.

2. **Link Them with a Trusted Source:** Link music-specific terms to a trusted external source or documentation. This could involve finding a centralized music schema or database that defines these terms or referencing the documentation of the software used to generate the data (e.g., jSymbolic for Jsymbolic extracted features in simssadb). We can also create our own source if none of the existing ones suffices. 

3. **Combination Approach:**  We would only link key music features that are significant for musicologists and are common across databases. Less common or specialized features may be treated as regular JSON keys with no external context. Additionally, we may choose not to store the later type of features at all if they are deemed unnecessary or non-metadata.

4. **Create the keys on wikidata anyway** 

Would love to hear what everyone thinks about this topic and please let me know if I misunderstood any part of the process. Thank you very much.

**Comment by dchiller** [comment]
First, I'll just post the like to WikiData's help page on [property creation](https://www.wikidata.org/wiki/Wikidata:Property_creation), which I think would be the procedure we'd go through for (4)...just to note that property creation is more complicated (slightly? significantly? it's hard to tell) than item creation. (Maybe everyone already knows this, but anyway...)

I think we should probably do 2, and for obscure properties that might only be relevant for a single database (and have the relevant database provide the iri?). So, as you say, have a jSymbolic feature reference jSymbolic. I guess the downside is that we won't be able to use wikidata entirely as the schema if we are ever indexing those "obscure" elements for search.

**Reply by malajvan** [reply]
I agree, I also think we've established that we won't dump everything on wikidata and uploading wikidata info is a problem we'll look into afterwards. During last Friday meeting @fujinaga also mentioned that we might/will create our own source for those obscure elements. For now at least for the current simssadb and cantusdb dumps, I've linked them with the closest resembling schema.org entry or wikidata entry for initial testing purposes.

**Reply by fujinaga** [reply]
I'm tending towards **#3 Combination Approach,** where we use Wikidata or Schema.org for any well-established concepts and terms. We will only add to Wikidata if these well-established concepts or terms are missing (e.g., name of instrument in a certain language). I think we should always store the original terms that were in each of the client databases. One of the reasons I like Linked Data is that we avoid language-specific expressions. I'm not so worried about not being able to use Wikidata entirely as the schema. I'm not even sure if we need to have complete schema as we are aiming somewhere between full indexing and full-text search. Don't forget we are performing lazy searching, meaning we do minimal processing (preserving original information) before we dump things into the Data Lake, then be smart, using NLP / machine learning, about how we do the search.

**Comment by malajvan** [comment]
From the meetings, this seems to be the general agreement that I'll apply to subsequent databases. We will find the most appropriate context for those specific keys by going through a list of sources ordered by priority (highest to lowest):
1. Wikidata
2. Schema.org
3. the databases themselves if they have the definition
4. (very rarely) other trusted source
The sources at number 4 must be long-lasting and trust-worthy
